java/src/jmri/jmrix/dccpp/DCCppReply.java

```diff
@@ -264,6 +265,92 @@ public String toMonitorString(){
         return text;
     }
 
+    /**
+     * Generate comment text for certain replies, used when objects are created from the message
+     *
+     * @return comment text
+     **/
+    public String toComment(){
+        StringBuilder text = new StringBuilder();
+        StringBuilder comma = new StringBuilder();
+        switch (getOpCodeChar()) {
+            case DCCppConstants.TURNOUT_REPLY:
+            case DCCppConstants.SENSOR_REPLY:
+            case DCCppConstants.OUTPUT_REPLY:
+                //write out properties in comment
+                getProperties().forEach((key, value) -> {
+                    text.append(comma).append(key).append(":").append(value);
+                    comma.setLength(0);
+                    comma.append(",");
```

Comment:
I suggest you change these lines to:
`text.append(',').append(key).append(':').append(value);`

and remove the variable `comma`.

Reply:
That starts the string with a comma, which I don't want.

Reply:
A more efficient code that removes the first comma is:
```
                boolean first = true;
                getProperties().forEach((key, value) -> {
                    if (!first) text.append(',');
                    text.append(key).append(":").append(value);
                    first = false;
                 });
```
Operating on a boolean is much faster than operating on a StringBuilder. Even if that probably doesn't matter in this case.

Reply:
That won't work with the lambda. Error: "Local variable first defined in an enclosing scope must be final or effectively final".

Reply:
I apologize. I missed that.